src/pattern-matching/exercise.rs

```diff
@@ -97,7 +97,7 @@ fn test_recursion() {
         left: Box::new(Expression::Op {
             op: Operation::Sub,
             left: Box::new(Expression::Value(3)),
-            right: Box::new(Expression::Value(4)),
+            right: Box::new(Expression::Value(0)),
```

Comment:
3-0 = 3, so this isn't a very interesting case of subtraction.

I guess the idea here is to uncover the bug where _any_ expression with 0 in the RHS is considered an error? I'm not convinced that's worth covering in an exercise like this, but maybe it could be done in another test case, rather than modifying this one.

Reply:
Yeah, that was the idea. I'm also not sure if we needs this tests, but it has come up during an real world session...
Added as separate test, but if that makes this file too long, perhaps it is not worth adding them.
In that case: no hard feelings, just close this pr...